libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -735,8 +735,19 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 		}
 		break;
 	}
+	case FunctionType::Kind::Creation:
+	{
+		auto const contractType = dynamic_cast<ContractType const*>(functionType->returnParameterTypes().front());
+		cout << contractType->contractDefinition().constructor()->name() << endl;
+		m_code <<
```

Comment:
This is more complicated - please take a look at ExpressionCompiler.cpp:625.

You need to:
Use `codecopy` to copy the object of the contract to memory. Then abi-encode the constructor arguments and append them. Then use the `create` or `create2` opcode to actually create the contract.

To test the code-generation for the constructor, you don't really need this, since you can call the constructor in tests directly:
```
// constructor(): 1, 2, 3, 4 ->
```